e2e/components/SelectPanel.test.ts

```diff
@@ -76,27 +76,47 @@ test.describe('SelectPanel', () => {
     })
   }
 
-  test(`Default @vrt forced-colors .modern-action-list--true`, async ({page}) => {
+  test(`Default @vrt forced-colors light .modern-action-list--true`, async ({page}) => {
     await visit(page, {
       id: 'components-selectpanel--default',
       globals: {featureFlags: {primer_react_select_panel_with_modern_action_list: true}},
     })
 
+    // windows high contrast mode: light
+    await page.emulateMedia({forcedColors: 'active', colorScheme: 'light'})
+    await page.evaluate(() => matchMedia('(prefers-color-scheme: light)').matches)
+
     // Open select panel
-    const isPanelOpen = await page.isVisible('[role="listbox"]')
+    const isPanelOpen = await page.getByRole('listbox').isVisible()
     if (!isPanelOpen) {
       await page.keyboard.press('Tab')
       await page.keyboard.press('Enter')
     }
+    await expect(page.getByRole('listbox')).toBeVisible()
 
-    // windows high contrast mode: light
-    await page.emulateMedia({forcedColors: 'active', colorScheme: 'light'})
     expect(await page.screenshot({animations: 'disabled'})).toMatchSnapshot(
       `SelectPanel-Default-forced-colors-light-modern-action-list--true.png`,
     )
+  })
+
+  test(`Default @vrt forced-colors dark .modern-action-list--true`, async ({page}) => {
```

Comment:
Related job: https://github.com/primer/react/actions/runs/12168939147/job/33940788592?pr=5356

The test was previously failing because the list panel closes when switching between page emulations from light to dark.

![Screenshot 2024-12-04 at 1 04 14 PM](https://github.com/user-attachments/assets/c6ce52eb-dede-4cf0-b47f-30b929fdf8df)

I was able to somewhat replicate the behavior locally (table below) and it seems to be related to the [toggled BaseStyles component for the feature flag](https://github.com/primer/react/pull/5356/files#diff-75e1d3dfdf55e8e4080753ed0a0a16c6b141f0c194d3e54010b4ebf26d6d2c74R55). I wasn't able to find a workaround and since it was just a page emulation for a test, I ended up simply splitting the tests and waiting for the list panel to open after `page.emulateMedia`

| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/231b84bc-9f19-4fdd-8e09-972af7bbb1f8" /> | <video src="https://github.com/user-attachments/assets/3f7c6d1f-39d7-47dc-91d6-eead6f2d21c5" /> |

Reply:
[The integration tests are failing]( https://github.com/github/github/pull/353683) which might be related to this. I'll investigate further tomorrow

Reply:
Found bug. The `toggleStyledComponent` needed to be called outside of the render method